packages/expo-updates/android/src/main/java/expo/modules/updates/statemachine/UpdatesStateMachine.kt

```diff
@@ -106,25 +107,29 @@ class UpdatesStateMachine(
           checkError = null,
           latestManifest = null,
           isUpdateAvailable = false,
-          isRollback = false
+          isRollback = false,
+          lastCheckForUpdateTime = Date(),
```

Comment:
On how this will be used, do you think users will expect it to be set right after they call `check` or upon getting the result back from the server? More concretely, should we set this for the `UpdatesStateEvent.Check` event rather than the results?

Reply:
My feeling was that the time should be set when the client gets the response... in practice it shouldn't make much difference.

Reply:
An alternative is to do what we already have in `UpdatesLogger`, where we convert to milliseconds since 1/1/70 and record that as a number. That gets put into the `timestamp` field of the updates logs.

Reply:
I imagine users will use this to conditionally display a button based on whether this was long ago enough. In that case, should that button disappear upon click or disappear when the response comes back? Or do we expect people to control button state on their own?

Reply:
I would think appearance and disappearance would be controlled by a useEffect on the last check time -- that would be a great thing to add to the demo app to see what would work the best. I'd want the button to gracefully animate away and not disappear suddenly when clicked :)

Reply:
Got it, yeah, that makes sense.